### ISSUE TITLE
Compiler definitions should be escaped to provide the same result on all actions

### ISSUE THREAD

**Issue by Dentoid** [bkaradzic/GENie#332]
Currently it seems like different generators (e.g. vstudio and gmake) handle escaping somewhat differently. For example, let's say I have a define line this:

`defines { 'SOME_DEFINE="Hello"' }`

In Visual Studio, this seems to be printed xml encoded if needed into the solution, and VS escapes this for the compiler so the actual code sees `"Hello"` as the value of `SOME_DEFINE`.
In gmake, however, it seems like it's written verbatim into the makefile and then eaten by the shell, so the code sees `Hello` as the value of `SOME_DEFINE`.

Exact specifics of how the data flows through the projects/makefiles aside, I would expect the final result in the code to match exactly what is specified in the genie-file with no manual escaping needed (except what's needed for entering the string in Lua of course).

This is not really as much a request for fixing this right away since I'm guessing it's behaving a bit different in different targets, actions, etc. and changing it might even be breaking existing geniefiles. It's more of a request for confirmation/agreement that this is how it should work, and any place where it doesn't work like this (like the gmake action for example) should be considered a bug to be fixed.

**Comment by JCash** [comment]
I'd like to upvote this issue as well, as I haven't yet managed to find a workaround for the problem.
I'm admittedly very new to Genie, so I may have missed something obvious.

In my case (generating XCode projects), my define is specified as:
```lua
    defines {
        'DLIB_LOG_DOMAIN=\"HID\"'
        "GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED",
        "GL_SILENCE_DEPRECATION",
    }
```
(and several variations of escaping)

But in the .pbxproj it ends up as:
```
				GCC_PREPROCESSOR_DEFINITIONS = (
					"DLIB_LOG_DOMAIN=\"HID\"",
					GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED,
					GL_SILENCE_DEPRECATION,
				);
```

And when building, it puts the entire thing as a string: `clang ´-DDLIB_LOG_DOMAIN=HID´` and it removes any escaping.
Manually changing the .pbxproj to `DLIB_LOG_DOMAIN=\"HID\"` works.

Is there a knows workaround for this?
I've tried using `buildoptions` but it seems to suffer from the same problem.

Regards,
Mathias

**Comment by bkaradzic** [comment]
> EDIT: I just saw this PR that seem to address the XCode part of it:

Can you test with this PR? Does it solve your issue?

**Comment by JCash** [comment]
Yes, I tried it and it does solve the issue, but unforunately breaks the xcode15 action, not sure why (it's not immediately obvious to be even though it's a quite small pr)
https://github.com/bkaradzic/GENie/pull/562/files#r1986257221
